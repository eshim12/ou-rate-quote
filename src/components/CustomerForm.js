import React, {useState} from 'react'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import Col from 'react-bootstrap/Col'

const CustomerForm = ({handleSubmit}) => {
	const [loanSize, setLoanSize] = useState(null)
	const [propertyType, setPropertyType] = useState(null)
	const [creditScore, setCreditScore] = useState(null)
	const [occupancy, setOccupancy] = useState(null)

	const propertyTypes = [
		{name: "Single Family", value: "SingleFamily"},
		{name: "Condo", value: "Condo"},
		{name: "Townhouse", value: "Townhouse"},
		{name: "Multi-Family", value: "MultiFamily"},
	]

	const occupancyTypes = [
		{name: "Primary", value: "Primary"},
		{name: "Secondary", value: "Secondary"},
		{name: "Investment", value: "Investment"},
	]

	return (
		<div>
			<Form onSubmit={(e) => handleSubmit(e, {loanSize, propertyType, creditScore, occupancy})}>
				<Form.Row>
					<Form.Group controlId='loan-size' as={Col}>
						<Form.Label>Loan Size</Form.Label>
						<Form.Control
							required
							type="number"
							placeholder="Loan Size"
							onChange={(e) => setLoanSize(e.target.value)}
						/>
					</Form.Group>
					<Form.Group controlId='property-type' as={Col}>
						<Form.Label>Property Type</Form.Label>
						<Form.Control as="select" required onChange={(e) => setPropertyType(e.target.value)}>
							<option></option>
							{propertyTypes.map((type, i) => <option key={i} value={type.value}>{type.name}</option>)}
						</Form.Control>
					</Form.Group>
				</Form.Row>
				<Form.Row>
					<Form.Group controlId='credit-score' as={Col}>
						<Form.Label>Credit Score</Form.Label>
						<Form.Control
							required
							type="number"
							placeholder="Credit Score"
							min="300"
							max="850"
							onChange={(e) => setCreditScore(e.target.value)}
						/>
					</Form.Group>

					<Form.Group controlId='occupancy' as={Col}>
						<Form.Label>Occupancy</Form.Label>
						<Form.Control as="select" required onChange={(e) => setOccupancy(e.target.value)}>
							<option></option>
							{occupancyTypes.map((type, i) => <option key={i} value={type.value}>{type.name}</option>)}
						</Form.Control>
					</Form.Group>
				</Form.Row>
				<Button variant="outline-primary" type="submit">Quote Rates</Button>
			</Form>
		</div>
	)
}

export default CustomerForm