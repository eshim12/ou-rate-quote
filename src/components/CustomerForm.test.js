import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import CustomerForm from './CustomerForm';

describe('CustomerForm', () => {
	test('snapshot renders', () => {
		const component = renderer.create(<CustomerForm/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})

describe('Fill inputs in CustomerForm', () => {
	let component;

    beforeEach(() => {
    	component = render(
			<CustomerForm/>
		)
    });

    test('input fields filled correctly', () => {
    	const params = {propertyType: 'Condo', creditScore: 700, loanSize: 200000, occupancy: 'Primary'}

    	const creditScoreInput = component.find("#credit-score")
    	creditScoreInput.value = params.creditScore
    	expect(component.find('#credit-score').length).toBe(1)

    	const proprtyTypeInput = component.find('#property-type')
    	proprtyTypeInput.value = params.proprtyType
    	expect(component.find('#property-type').length).toBe(1)

    	const loanSizeInput = component.find('#loan-size')
    	proprtyTypeInput.value = params.loanSize
    	expect(component.find('#loan-size').length).toBe(1)

    	const occupancyInput = component.find('#occupancy')
    	proprtyTypeInput.value = params.occupancy
    	expect(component.find('#occupancy').length).toBe(1)
    })
})