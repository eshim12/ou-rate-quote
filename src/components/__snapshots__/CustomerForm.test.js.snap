// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerForm snapshot renders 1`] = `
<div>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="loan-size"
        >
          Loan Size
        </label>
        <input
          className="form-control"
          id="loan-size"
          onChange={[Function]}
          placeholder="Loan Size"
          required={true}
          type="number"
        />
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="property-type"
        >
          Property Type
        </label>
        <select
          className="form-control"
          id="property-type"
          onChange={[Function]}
          required={true}
        >
          <option />
          <option
            value="SingleFamily"
          >
            Single Family
          </option>
          <option
            value="Condo"
          >
            Condo
          </option>
          <option
            value="Townhouse"
          >
            Townhouse
          </option>
          <option
            value="MultiFamily"
          >
            Multi-Family
          </option>
        </select>
      </div>
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="credit-score"
        >
          Credit Score
        </label>
        <input
          className="form-control"
          id="credit-score"
          max="850"
          min="300"
          onChange={[Function]}
          placeholder="Credit Score"
          required={true}
          type="number"
        />
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="occupancy"
        >
          Occupancy
        </label>
        <select
          className="form-control"
          id="occupancy"
          onChange={[Function]}
          required={true}
        >
          <option />
          <option
            value="Primary"
          >
            Primary
          </option>
          <option
            value="Secondary"
          >
            Secondary
          </option>
          <option
            value="Investment"
          >
            Investment
          </option>
        </select>
      </div>
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      type="submit"
    >
      Quote Rates
    </button>
  </form>
</div>
`;
